Source/Dafny/Verifier/Translator.cs

```diff
@@ -8889,6 +8905,65 @@ private class WFOptions
       return new Bpl.IdentifierExpr(tok, "$OneHeap", predef.HeapType);
     }
 
+    /// <summary>
+    /// For expression "e" that is expected to come from a modifies/unchanged frame, return information
+    /// that is useful for checking every reference from "e". More precisely,
+    ///  * If "e" denotes a reference, then return
+    ///       -- "description" as the string "object",
+    ///       -- "type" as the type of that reference,
+    ///       -- "obj" as the translation of that reference, and
+    ///       -- "antecedent" as "true".
+    ///  * If "e" denotes a set of references, then return
+    ///       -- "description" as the string "each set element",
+    ///       -- "type" as the element type of that set,
+    ///       -- "obj" as a new identifier of type "type", and
+    ///       -- "antecedent" as "obj in e".
+    ///  * If "e" denotes a sequence of references, then return
+    ///       -- "description" as the string "each sequence element",
+    ///       -- "type" as the element type of that sequence,
+    ///       -- "obj" as an expression "e[i]", where "i" is a new identifier, and
+    ///       -- "antecedent" as "0 <= i < |e|".
+    /// </summary>
+    void EachReferenceInFrameExpression(Expression e, List<Bpl.Variable> locals, BoogieStmtListBuilder builder, ExpressionTranslator etran,
```

Comment:
The reason `fresh` supports these different types of arguments is because Dafny doesn't have a concept of varargs, right?

Writing `fresh(...{a,b,c})` over `fresh({a,b,c})` seems fine to me (`...` is just example syntax).

Reply:
I think of `fresh(S)` as the "real" syntax, where `S` is a set of objects. But it frequently happens that you want to mention just one object, and therefore it seems reasonable to support `fresh(x)` as a shorthand for `fresh({x})`.

One could indeed think of it the other way around, too. That is, you could think of `fresh(x)` as the "real" syntax, which case `fresh(S)` would be a shorthand for `forall x :: x in S ==> fresh(s)`.

So, the language could pick just one of these, but there are common uses of both, so supporting both seems nice.